File: domain/src/test/java/org/openmedstack/domain/ConventionEventRouterTest.java
package org.openmedstack.domain;

import openmedstack.events.BaseEvent;
import org.junit.Assert;
import org.junit.Test;

import java.time.Instant;

public class ConventionEventRouterTest {
    @Test
    public void generatesHandlersFromHandlerSource() {
        var handler = new TestHandler();
        var router = new ConventionEventRouter(true, handler);
        try {
            router.dispatch(new TestMessage("test", Instant.now()));
        } catch (HandlerForDomainEventNotFoundException e) {
            Assert.fail();
        }
    }
}

class TestHandler{
    public void apply(TestMessage msg){
    }
}

class TestMessage extends BaseEvent{
    protected TestMessage(String source, Instant timeStamp) {
        super(source, timeStamp);
    }
}

File: domain/src/test/kotlin/org/openmedstack/domain/ConventionEventRouterTest.kt
package org.openmedstack.domain

import org.junit.Assert
import org.junit.Test
import org.openmedstack.events.BaseEvent
import java.time.Instant

class ConventionEventRouterTest {
    @Test
    fun generatesHandlersFromHandlerSource() {
        val handler = TestHandler()
        val router = ConventionEventRouter(true, handler)
        try {
            router.dispatch(TestMessage("test", Instant.now()))
        } catch (e: HandlerForDomainEventNotFoundException) {
            Assert.fail()
        }
    }
}

internal class TestHandler {
    fun apply(msg: TestMessage) {
        println(msg.source)
    }
}

internal class TestMessage(source: String, timeStamp: Instant) : BaseEvent(source, timeStamp)